dataactbroker/handlers/accountHandler.py

```diff
@@ -884,3 +865,30 @@ def email_users(self, system_email, session):
             new_email.send()
 
         return JsonResponse.create(StatusCode.OK, {"message": "Emails successfully sent"})
+
+
+def grant_superuser(user):
+    user.cgac_code = 'SYS'
+    user.website_admin = True
+    user.permission_type_id = PERMISSION_TYPE_DICT['writer']
+
+
+def grant_highest_permission(user, group_list, cgac_group):
+    """Find the highest permission within the provided cgac_group; set that as
+    the user's permission_type_id"""
+    user.cgac_code = cgac_group[-3:]
+    user.website_admin = False
+    permission_group = [g for g in group_list
+                        if g.startswith(cgac_group + "-PERM_")]
+    # Check if a user has been placed in a specific group. If not, deny access
+    if not permission_group:
+        user.permission_type_id = None
+    else:
+        perms = [perm[-1].lower() for perm in permission_group]
+        ordered_perms = sorted(
+            PERMISSION_MAP.items(), key=lambda pair: pair[1]['order'])
+        for key, permission in ordered_perms:
+            name = permission['name']
+            if key in perms:
+                user.permission_type_id = PERMISSION_TYPE_DICT[name]
+                break
```

Comment:
Or, if they're write for one and read for another, I assume they can only do read for the one they're read in, so how is that going to work?